src/BlazorWebView/src/Maui/Windows/WinUIWebViewManager.cs

```diff
@@ -21,13 +22,25 @@ public class WinUIWebViewManager : WebView2WebViewManager
 		private readonly WebView2Control _webview;
 		private readonly string _hostPageRelativePath;
 		private readonly string _contentRootDir;
+		private readonly ExternalLinkMode _externalLinkMode;
 
-		public WinUIWebViewManager(WebView2Control webview, IServiceProvider services, Dispatcher dispatcher, IFileProvider fileProvider, JSComponentConfigurationStore jsComponents, string hostPageRelativePath, string contentRootDir)
+		public WinUIWebViewManager(
+			WebView2Control webview,
+			IServiceProvider services,
+			Dispatcher dispatcher,
+			IFileProvider fileProvider,
+			JSComponentConfigurationStore jsComponents,
+			string hostPageRelativePath,
+			string contentRootDir,
+			ExternalLinkMode externalLinkMode)
 			: base(webview, services, dispatcher, fileProvider, jsComponents, hostPageRelativePath)
 		{
 			_webview = webview;
 			_hostPageRelativePath = hostPageRelativePath;
 			_contentRootDir = contentRootDir;
+			_externalLinkMode = externalLinkMode;
+
+			_webview.CoreWebView2Initialized += Webview_CoreWebView2Initialized;
```

Comment:
Does this need to unsubscribed when this type is disposed?

Reply:
I was under the impression the lifecycle of the `WinUIWebViewManager` and the associated `WebView2Control` would be essentially the same, hence I didn't bother with unsubscribing (given if they get disposed at the same time then there shouldn't be any stray references preventing GC). This is in line with the current async lambda we have for starting the blazor script:

https://github.com/dotnet/maui/blob/b00586cc0dd5dae880a5ebdbdd467c6f0663b87b/src/BlazorWebView/src/Maui/Windows/WinUIWebViewManager.cs#L98-L103


I can add it in if you think it'd be beneficial.

Reply:
Alternatively, instead of hooking the Initialized event at all, you can wait until we force initialization and then hook the events you actually want directly. Here's where we force initialization in the shared code: https://github.com/dotnet/maui/blob/main/src/BlazorWebView/src/SharedSource/WebView2WebViewManager.cs#L93

Because presumably we want exactly this logic on WinForms and WPF as well. Launching the browser will be different between WinUI and non-WinUI, but that could be abstracted via a delegate or something.